### PR TITLE
Prohibit changing of transition_aged_youth

### LINKED ISSUE #896: remove ability to edit transition_aged_youth
**Description**
Previously, it was up to users to mark a `casa_case` as `transition_aged_youth`. After #831 is done, this will be automated. So users no longer need the ability to edit this manually. This change should be made on the frontend _and_ the backend so that there is only one trigger that changes `transition_aged_youth` from `false` to `true`. 

**Where does/should this occur?**
On the Edit CASA Case page accessible by all users. (<a href="https://casa-qa.herokuapp.com/casa_cases/152/edit">Here is an example</a>. Login using: email - supervisor1@example.com // pw: 123456)

**How is it working now?**
All users can change `transition_aged_youth` from `false` to `true` and vice versa. 

**How should it work instead?**
No user should have the ability to adjust `transition_aged_youth` status for any `casa_case`.

### PR DESCRIPTION
**Pull request by rafaeelaudibert**
### What github issue is this PR for, if any?
Resolves #896 

### What changed, and why?
Removed the ability to change the `transition_aged_youth` flag. I still allowed the field to be used when creating the case.

### CODE REVIEW

File: app/views/casa_cases/_form.html.erb
Comment (littleforest) [inline app/views/casa_cases/_form.html.erb:23]: I think it's easier to not read negatives (despite the structure of this sentence) 😬 :

```
<% if casa_case.new_record? %>
```

**Review by littleforest — CHANGES_REQUESTED** [review]

File: app/views/casa_cases/_form.html.erb
Comment (rafaeelaudibert) [reply app/views/casa_cases/_form.html.erb]: You are absolutely right. Changed it!

**Comment by jmkoni** [comment]
Downside is  that this is dependent on #831, so we want to get a fix for that before this is merged in. But it does look good!

**Review by compwron — APPROVED** [review]
Looks good! I agree that we need to get #831 done but I don't want to hold this up for fear of merge conflicts. @rafaeelaudibert if you have time, do you want to take on #831 also? :)

**Comment by rafaeelaudibert** [comment]
Hey Guys! Sorry for the delay! I could have tackled that issue, but as I see, it has already been closed. Will take a look on another open issue to give it a go! Thanks for the merge

**Comment by compwron** [comment]
You are invited to the CASA asynchronous retrospective-and-futurespective! Please take a look at https://bit.ly/casaretro (no login required) and add at least one note! Even if it's just "I was here and you are cool" or "omg y'all why do you keep messaging me" we want to hear it. <3
Next week Wednesday 18 November @ 6pm Pacific time we will Zoom about the board and talk about themes and takeaways. (Zoom link in our calendar at http://bit.ly/casacal or slack #casa channel https://rubyforgood.herokuapp.com/ )
Please ESPECIALLY add lots of notes if you won't be able to make it on Wednesday to talk!
* Note: the board will be monitored for spam; message me or any team lead if you see a problem
https://bit.ly/casaretro

**Pull request merged**
